pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -378,6 +378,12 @@ func (e *Controller) syncService(ctx context.Context, key string) error {
 		return nil
 	}
 
+	if service.Spec.Type == v1.ServiceTypeExternalName {
+		// services with Type ExternalName receive no endpoints from this controller;
+		// Ref: https://issues.k8s.io/105986
+		return nil
+	}
```

Comment:
I believe this would also not result in Endpoints getting cleaned up after a Service type changes, not sure if that's what we want.

Reply:
Replied in the other comment